snapshot-labs/sx-monorepo
# feat: add support for starknet account to offchain actions

### Summary



Closes: #372 and https://github.com/snapshot-labs/pitches/issues/83

This PR add starknet wallet support for `setAlias` on offchain networks.

Once alias is created, a starknet account will be able to follow/unfollow a space and edit its user profile

With this PR, features on starknet accounts will be on par with EVM accounts

### Changes 

- feat: create an alias from a starknet wallet
- fix: enable back follow/unfollow and updateUser button for starknet users
- fix: load followed spaces for starknet users
- fix: update follow/unfollow/updateUsers type ons sx/offchain to accept starknet address
- fix: update starknet-sig returned envelop to always return `address` as a padded address
- fix: do not load starknet user's vote from offchain hub
- refactor: remove `SignatureData` from sx/offchain/types, and use the global one from sx/types

### How to test

~~Waiting merging of https://github.com/snapshot-labs/snapshot-sequencer/pull/400~~

- Log in with a starknet wallet 
- Follow a space
- It should ask to sign a transaction to create an alias
- After signing, it should follow the space, and the space should appear on the sidebar
- Refresh the page
- The followed space should appear in the sidebar
- You can follow/unfollow more spaces (offchain/evm/starknet spaces)
- Go to your user profile
- The EDIT button should be visible
- Edit and save your profile
- It should save and show your updated profile
